astropy/io/votable/tests/test_vo.py

```diff
@@ -276,7 +279,45 @@ def test_select_columns_by_name():
     assert array["string_test"][0] == "String & test"
     for c in columns:
         assert not np.all(mask[c])
-    assert np.all(mask["unicode_test"])
+
+    # deselected columns shouldn't be present in the output
+    assert "unicode_test" not in array.dtype.fields
+    assert "unicode_test" not in mask.dtype.fields
+
+
+@pytest.mark.parametrize(
+    "column_ids, use_names_over_ids, expected_names",
+    [
+        # just the first column
+        (["string_test"], False, ["string_test"]),
+        (["string_test"], True, ["string test"]),
+        # a single column, other than the first
+        (["unicode_test"], False, ["unicode_test"]),
+        (["unicode_test"], True, ["unicode_test"]),
+        # just the first two columns (that have the same ID)
+        (["string_test", "string_test_2"], False, ["string_test", "string_test_2"]),
+        (["string_test", "string_test_2"], True, ["string test", "fixed string test"]),
+        # two non-consecutive, differently named columns
+        (["string_test", "unicode_test"], False, ["string_test", "unicode_test"]),
+        (["string_test", "unicode_test"], True, ["string test", "unicode_test"]),
+    ],
+)
+def test_select_columns_by_name_edge_cases(
+    column_ids, use_names_over_ids, expected_names
+):
+    # see https://github.com/astropy/astropy/issues/14943
+    filename = get_pkg_data_filename("data/regression.xml")
+    vot1 = parse_single_table(filename, columns=column_ids)
+    t1 = vot1.to_table(use_names_over_ids=use_names_over_ids)
+    colnames1 = [name for name in t1.columns]  # noqa: C416
+    assert colnames1 == expected_names
```

Comment:
Doesn't the following work?
```suggestion
    assert t1.colnames == expected_names
```

Reply:
it does, thanks ! I forgot about `Table.colnames`